build_tag_push.py

```diff
@@ -1,101 +1,130 @@
-	print image_id
-	#os.environ['IMAGE_ID'] = image_id  # This -q flag retrieves the last image ID (Doesn't work... TypeError)
-	#os.system("docker run $IMAGE_ID")
-	#os.system("docker run docker images -q $IMAGE_NAME") -> Nope.
-	#os.system("docker images -q $IMAGE_NAME")
-	#os.system("IMAGE_ID=$(docker images -q $IMAGE_NAME)")
-	os.system("echo done")
-	#os.system('docker run $IMAGE_ID')
-
-	##########################################
-	# NOT FINISHED HERE -- WILL PICK UP ON IT.
-	# Left off on trying to store the Image ID 
-	# as an environment variable to use that when 
-	# running the ID. 
-	##########################################
-
-
-# Step 4 - Tags the image with the current commit hash & pushed to Docker Hub = NOT COMPLETE.
-#
-#
-#
-#
-#
-#
-#
-#
-#
-#
-#
+
+def build_DockerImage(filename):
+	docker_dir_context = createContextDirectory(parseAppFile(filename)['docker_context']) # This will be the build and static file (and needs a Dockerfile as well)
+	image_name = parseAppFile(filename)['name'] # Name from app.yml file
+	subprocess.check_call(["docker", "build", docker_dir_context, "-t", image_name]) # this builds the docker image.
+
+	# QUESTION: When I try to build the image from the file that has the docker context from the app.yml file 
+	# then I get the error: "Build an image from a Dockerfile". The only way I was able to complete this was 
+	# putting in my own Dockerfile, which I am assuming the groups will have as well and we will need to copy
+	# this over into that directory created, or is this going to be copied over manually?
+
+# Step 4 - Tags the image with the current commit hash & push to Docker Hub = COMPLETE.
+
+def tag_DockerImage(filename):
+	'''
+	This function is finding the tagging the Docker Image with the name of the user/image_name, and the current git hash. 
+	Note: The current git hash is shortened a bit, but to display the full hash, then change the lower case %h to upper case %H. 
+	The source for this command is: https://container-solutions.com/tagging-docker-images-the-right-way
+	'''
+	image_name = parseAppFile(filename)['name']
+
+	subprocess.check_call(["docker", "tag", image_name, Docker_Hub_Name+'/'+image_name]) # Need to tag the image with the namespace for the Docker Hub account. 
+																						 # Note: I made this a global variable for now with a comment for user input if needed.
+
+	current_hash = str(subprocess.check_call(["git", "log", "-1", "--pretty=%h"])) # Note: having the lower case 'h' provides a shorter version of the hash... 
+	 																			   # I read that the shorter tag was better so I did that here.
+
+	subprocess.check_call(["docker", "tag", image_name, Docker_Hub_Name+'/'+image_name+':'+current_hash]) # This tags the image with the current hash.
+
+	# QUESTION: When I am running this function, the current_hash is turning to the number 0, and not what the actual current_hash is when observed in the terminal (which is 07f21e2), 
```

Comment:
that's because `check_call` doesn't return the stdout of the process you just ran. You need `subprocess.check_output` which does. Note that `git log` will add a newline at the end of the hash so you'll probably need to call `strip` on the string you get back from `check_output`

If you don't know what stdout, stdin, and stderr are we should talk about that when we meet today.